src/Strategy/Python/PyProjectGeneric.hs

```diff
@@ -0,0 +1,501 @@
+{-# LANGUAGE TemplateHaskell #-}
```

Comment:
We already have a `pyproject.toml` parser in `src/Strategy/Python/Poetry/PyProject.hs`. The intent of this ticket wasn't to create a new one: it was to reify that tactic as its own strategy which augmented its responses with `Poetry/PDM` information. 

As a type you could imagine something like this:

```haskell
data PyProject buildSystem = PyProject {
  commonField :: ...,
  commonField2 :: ...,
  projectSpecificInfo :: buildSystem
}
```
^^ Though you don't need to do it this way.

I'm pretty uncomfortable with merging in a totally new pyproject parser unless it's obviously better than what we have. Even in that case we should deprecate the old one. What is your sense here/do you have thoughts about that?